mujoco_warp/_src/collision_driver.py

```diff
@@ -495,16 +520,22 @@ def _narrowphase(m, d):
     sdf_narrowphase(m, d)
 
 
-def narrowphase(m, d):
-  if m.opt.graph_conditional:
-    wp.capture_if(condition=d.ncollision, on_true=_narrowphase, m=m, d=d)
-  else:
-    _narrowphase(m, d)
-
-
 @event_scope
 def collision(m: Model, d: Data):
-  """Collision detection."""
+  """Runs the full collision detection pipeline.
+
+  This function orchestrates the broadphase and narrowphase collision detection stages. It
+  first identifies potential collision pairs using a broadphase algorithm (either N-squared
+  or Sweep-and-Prune, based on `m.opt.broadphase`). Then, for each potential pair, it
+  performs narrowphase collision detection to compute detailed contact information like
+  distance, position, and frame.
+
+  The results are used to populate the `d.contact` array, and the total number of contacts
```

Comment:
we might want to note that `ncon` can be greater than the total number of contacts